src/registrar/models/domain_application.py

```diff
@@ -501,6 +501,18 @@ def _send_status_update_email(
         except EmailSendingError:
             logger.warning("Failed to send confirmation email", exc_info=True)
 
+    def set_approved_domain(self):
+        """Always called in parallel to the approve method"""
+
+        # The domain should be ready for us, let's find it
+        Domain = apps.get_model("registrar.Domain")
+        if not Domain.objects.filter(name=self.requested_domain.name).exists():
+            raise ValueError("Something went wrong and the domain did not get created.")
+        created_domain = Domain.objects.get(name=self.requested_domain.name)
```

Comment:
created_domain implies it's being made, and get just retrieves. maybe linked_domain here and below (or something else)